### ISSUE TITLE
Nginx Reverse Proxy for UI

### ISSUE THREAD

**Issue by russokiwi** [its-a-feature/Mythic#116]
Sorry to dig this up again guys.  So if I have my primary Nginx instance listening on 443 and passing all to mythic_nginx listening on 10443, like so::

location / {
        proxy_pass http://127.0.0.1:10443;
        proxy_pass_request_headers on;
        proxy_pass_request_body on;
        proxy_http_version 1.1;
        proxy_redirect off;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header X-Forwarded-Port $server_port;
    }

There's still no websocket connect unfortunately / can't use the UI...  Please help..

**Comment by pkudrel** [comment]
I had similar issue. In my case that helped:
1. Update `http` block in `nginx.conf`

```
http {
...
 map $http_upgrade $connection_upgrade {
        default upgrade;
        '' close;
    }
...
}
```

2. Update site config 
```
    location / {
                 proxy_pass https://127.0.0.1:10443/;
                 proxy_set_header Connection $connection_upgrade;
                 proxy_set_header Host $host;
                 proxy_set_header X-Forwarded-Proto $scheme;
                 proxy_set_header X-Forwarded-Port $server_port;
                 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                 proxy_http_version 1.1;
                 proxy_set_header Upgrade $http_upgrade;
                 proxy_set_header X-Real-IP $remote_addr;
                 proxy_set_header X-Forwarded-Host $host;
                 proxy_set_header X-Forwarded-Server $host;
                 proxy_pass_request_headers on;
                 add_header X-location websocket always;
                 proxy_read_timeout 900s;

        }
```

**Comment by its-a-feature** [comment]
@russokiwi I think the trailing `/` at the end of the proxy_pass is important

**Comment by russokiwi** [comment]
Thanks so much for quick response guys, unfortunately not good still - yes totally noted re the trailing `/` - sorry was a typo.  So what is interesting is that the host server which is handling uri proxying(443) whilst hosting mythic_nginx (10443) can browse to it's own 443 fine without the websocket issue.  It's as soon as I hit it from any other node that everything breaks in the relay..

**Comment by its-a-feature** [comment]
Can you explain your setup and what you're trying to do again? I think it was initially in some other issue?

**Comment by eddiezab** [comment]
So I'm not a user of mythic (yet) we've just started evaluating it. But we were trying a similar deployment. In addition to what was done here https://github.com/its-a-feature/Mythic/issues/116#issuecomment-911833415 we've also added the directives to the location block

`
            sub_filter ':10443' '';
            sub_filter_once off;
`

It appears the UI works properly now for us. Of course, I'm not sure what other issues --if any -- this may cause
